include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -522,10 +522,16 @@ struct __get_sycl_range
     {
         assert(__first < __last);
         using value_type = val_t<_Iter>;
+
+        auto __offset = __first - oneapi::dpl::begin(__first.get_buffer());
+        auto __n = __last - __first;
+        auto __size = __first.get_buffer().size();
+        if (__offset + __n > __size)
+            __n = __size - __offset;
```

Comment:
I believe we don't need to "correct" size of sequence.. We should an additional assert here like `assert(__offset + __n <= __size)`.

Reply:
Fixed: changed to assert; fixed error in histogram test.